### DISCUSSION TITLE
Can arguments be supplied for extensions in config file?

### DISCUSSION THREAD

**Discussion by hwine** [pyscaffold/pyscaffold#678] [Q&A]
n00b to pyscaffold, but couldn't find this addressed. (I just learned about pyscaffold on [podcast.\_\_init__](https://www.pythonpodcast.com/pyscaffold-project-template-boilerplate-episode-384/))

I like using `pip-tools`, and would like the [suggested workflow](https://pyscaffold.org/en/stable/dependencies.html#integration-with-pip-tools) to be my default. However, I can't figure out how to enter the packages to be installed (`pip-tools` & `setuptools_scm`) into my local configuration (`~/.config/pyscaffold/defaults.cfg`).

Using the `--save-config` option does not record use of the `venv` extension. I can add the `venv` extension manually to the config file, and that works. But it doesn't automate the installation of the packages.

What's the pyscaffold-ic way to achieve this sort of local default behavior, short of a custom extension?

**Comment by abravalheri** [comment]
Hi @hwine, when I was writing the venv extension I considered that it would be better to make this extension "non persistent", that is the reason why it doesn't go to `defaults.cfg`.

You can add it manually to the file and it should create the virtualenv for you, for example:

```ini
[metadata]
author = John Doe
author-email = john.joe@gmail.com
license = MPL-2.0

[pyscaffold]
extensions =
    cirrus
    venv
```

In principle it would be possible also to add a `venv_install` option to the same file. However, there is a problem: `.cfg` files follow a INI grammar which does not understand the concept of lists/arrays of objects. Right now, PyScaffold would simply iterate through `venv_install` and try to install each char of the string as if it was a different package. I believe the next release would solve this problem by using `.toml` files instead.

I will have a look and see if we can do something about it before PyScaffold v5.

**Reply by hwine** [reply]
@abravalheri thanks for the confirmation. I don't think there is a need to solve this before 5.x. TOML support will help on many fronts. 

I'll just use a shell function for now. 😁
